### PR TITLE
feat: keep the css maintenance mode if any env resolves to passive ip

### PR DESCRIPTION
**Pull request by NithinKuruba**

### CODE REVIEW

File: src/clients/dns.py [hunk 1]
```diff
@@ -79,3 +79,30 @@ def is_keycloak_dr_up_and_receiving_traffic(ip: str):
...
+        "prod": "loginproxy.gov.bc.ca",
+    }
+    if prod:
+        return switcher.get(env, "")
+    else:
+        return switcher.get("%s-%s" % ("sandbox", env), "")
+
+
+def check_dns_by_env(env, mode):
+    print('hello')
```
Comment (thegentlemanphysicist) [inline src/clients/dns.py:100]: print hello

File: src/clients/dns.py [hunk 2]
```diff
@@ -79,3 +79,30 @@ def is_keycloak_dr_up_and_receiving_traffic(ip: str):
...
+    try:
+        ip_address = socket.gethostbyname(fetch_domain_by_env(env, "sandbox" not in config.get('domain_name')))
+        if(config.get(mode) == ip_address):
```
Comment (thegentlemanphysicist) [inline src/clients/dns.py:103]: did the config file get modified? my copy doesn't have 'mode' in it?

File: src/logic.py [hunk 3]
```diff
@@ -38,8 +39,19 @@ def handle_queues(queue: Queue, processes: list):
...
+        css_maintenance_to_active = True
+        for env in ['dev', 'test', 'prod']:
+            dns_matched = check_dns_by_env(env, passive_ip)
+            if(dns_matched or dns_matched == 'error'):
+                logger.info("%s is still pointing to %s or unable to check dns" % (env, passive_ip))
+                css_maintenance_to_active = False
+                break
+
+        if css_maintenance_to_active:
+            logger.info("active_ip")
```
Comment (thegentlemanphysicist) [inline src/logic.py:51]: I would put a more descriptive log here, maybe something like "All (sandbox/production) environments' traffic goin to Gold, setting CSS app to active"

File: src/clients/dns.py (hunk 1 above)
Comment (NithinKuruba) [reply src/clients/dns.py:100]: hahaha that got through some how. Removed it

File: src/clients/dns.py (hunk 2 above)
Comment (NithinKuruba) [reply src/clients/dns.py:103]: Mode is the function argument

File: src/logic.py (hunk 3 above)
Comment (NithinKuruba) [reply src/logic.py:51]: It is the default behaviour so we don't need to say it explicitly anyways. If there is any env still pointing to passive ip, we would know it from the log

**Review by thegentlemanphysicist — APPROVED** [review]

**Pull request merged**